node/sc/bridge_manager.go

```diff
@@ -311,7 +314,19 @@ func (bi *BridgeInfo) handleRequestValueTransferEvent(ev *RequestValueTransferEv
 		}
 		logger.Trace("Bridge succeeded to HandleERC20Transfer", "nonce", ev.RequestNonce, "tx", handleTx.Hash().String())
 	case ERC721:
-		handleTx, err = bi.bridge.HandleERC721Transfer(auth, ev.Raw.TxHash, ev.From, ev.To, tokenAddr, ev.ValueOrTokenId, ev.RequestNonce, ev.Raw.BlockNumber, ev.Uri, ev.ExtraData)
+		// get URI of the ERC721
+		var uri string
+		erc721, err := scnft.NewERC721Metadata(ev.TokenAddress, bi.counterpartBackend)
```

Comment:
Is it safe to call every time to handle events?

Reply:
I think it's safe. Because after minting NFT, owner can't change it in normal ERC721Metadata.

Reply:
How about caching the erc721 object not to recreate it for each event?

Reply:
I think there is not big processing during create ERC721 object. 
But, anyway can I consider it in next PR.